test/functional/wallet-hd.py

```diff
@@ -75,10 +75,12 @@ def run_test (self):
 
         self.log.info("Restore backup ...")
         self.stop_node(1)
-        os.remove(self.options.tmpdir + "/node1/regtest/wallet.dat")
+        # we need to delete the complete regtest directory
+        # otherwise node1 would auto-recover all funds in flag the keypool keys as used
```

Comment:
In commit "Ensure we pause wallet sync when the keypool requires extension"

Is this change meant to be part of this commit? I don't think there's an interaction with hdignoregaplimit here.